examples/personachat-example.ipynb

```diff
@@ -0,0 +1,317 @@
+    ")\n",
+    "\n",
+    "\n",
+    "tokenized_datasets.set_format(\"torch\")\n",
+    "train_dataset = tokenized_datasets[\"train\"].shuffle(seed=SEED)\n",
+    "train_dataloader = DataLoader(\n",
+    "    train_dataset.select(list(range(NUM_SAMPLES))),\n",
+    "    shuffle=True,\n",
+    "    batch_size=BATCH_SIZE,\n",
+    "    drop_last=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef4323fd",
+   "metadata": {},
+   "source": [
+    "Before setting up optimizers, check model parameters that will be trained."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9cc0ba34",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for n, p in model.named_parameters():\n",
+    "    if p.requires_grad:\n",
+    "        print(n, p.requires_grad, p.device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "59cffce7",
+   "metadata": {},
+   "source": [
+    "Optimizer will only work on **prompts**, they are only trainable parameters. So initialize optimizer and learning rate scheduler."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ef9bf344",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "optimizer = AdamW(model.parameters(), lr=LR, weight_decay=WEIGHT_DECAY)\n",
+    "\n",
+    "lr_scheduler = get_scheduler(\n",
+    "    name=\"linear\", optimizer=optimizer, num_warmup_steps=0, num_training_steps=len(train_dataloader)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "423c56d5",
+   "metadata": {},
+   "source": [
+    "Let's initialize wandb for logging and start training loop!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d9e46807",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wandb.init(\n",
+    "    project=\"bloom-personachat\",\n",
+    "    config={\n",
+    "        \"num_samples\": NUM_SAMPLES,\n",
+    "        \"batch_size\": BATCH_SIZE,\n",
+    "        \"learning_rate\": LR,\n",
+    "        \"weight_decay\": WEIGHT_DECAY,\n",
+    "        \"num_prefix_tokens\": NUM_PREFIX_TOKENS,\n",
+    "        \"model_name\": MODEL_NAME,\n",
+    "        \"seed\": SEED,\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "progress_bar = tqdm(range(len(train_dataloader)))\n",
+    "for batch in train_dataloader:\n",
+    "    batch = {k: v.to(DEVICE) for k, v in batch.items()}\n",
+    "\n",
+    "    model.train()\n",
+    "    outputs = model(**batch)\n",
+    "    loss = outputs.loss\n",
+    "    loss.backward()\n",
+    "\n",
+    "    optimizer.step()\n",
+    "    lr_scheduler.step()\n",
+    "    optimizer.zero_grad()\n",
+    "\n",
+    "    wandb.log({\"Train Loss\": loss})\n",
+    "    progress_bar.update(1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0f36cb80",
+   "metadata": {},
+   "source": [
+    "Try to talk with the trained model! To break from dialog mode, press Ctrl+D."
```

Comment:
```suggestion
    "Try to talk with the trained model! Submit an empty input to stop the execution."
```

I'm not sure if Ctrl+D works well on all platforms, but there's an easier way to stop this code.